packages/airnode-node/src/workers/cloud-platforms/aws.ts

```diff
@@ -1,49 +1,43 @@
-  // lambda.invoke is synchronous so we need to wrap this in a promise
-  return new Promise((spawnResolve, spawnReject) => {
-    // Uses the current region by default
-    const lambda = new AWS.Lambda();
-
-    // AWS doesn't allow uppercase letters in lambda function names
-    const resolvedName = `airnode-${params.deploymentId}-run`;
-
-    const options = {
-      FunctionName: resolvedName,
-      Payload: JSON.stringify(params.payload),
-    };
-
-    const resolve: typeof spawnResolve = (value) => {
-      logger.debug(`Successful Lambda response: ${value}`);
-      spawnResolve(value);
-    };
-
-    const reject: typeof spawnReject = (error) => {
-      logger.debug(`Lambda invocation or execution failed. Response: ${error}`);
-      spawnReject(error);
-    };
-
-    // The Lambda invocation callback populates the error (first argument) only when the invocation fails (e.g. status
-    // code is 400) or the request is not parsed properly by the SDK and can't be invoked. For errors and timeouts that
-    // happen inside the invoked lambda have "data.FunctionError" and "data.Payload.errorMessage" populated instead.
-    // See: https://stackoverflow.com/q/42672023 and https://stackoverflow.com/q/48644093
-    lambda.invoke(options, (err, data) => {
-      if (err) return reject(err);
-
-      if (data.FunctionError) return reject(data.Payload);
-
-      if (!data.Payload) return reject(new Error('Missing payload in lambda response'));
-      const goPayload = goSync(() => JSON.parse(data.Payload!.toString()));
-      if (!goPayload.success) return reject(goPayload.error);
-
-      const goBody = goSync(() => JSON.parse(goPayload.data.body));
-      if (!goBody.success) return reject(goBody.error);
-
-      resolve(goBody.data as WorkerResponse);
-    });
-  });
+export async function spawn(params: WorkerParameters): Promise<WorkerResponse> {
+  // Uses the current region by default
+  const awsLambdaClient = new LambdaClient({});
+
+  // AWS doesn't allow uppercase letters in lambda function names
+  const resolvedName = `airnode-${params.deploymentId}-run`;
+
+  const options = {
+    FunctionName: resolvedName,
+    Payload: new TextEncoder().encode(JSON.stringify(params.payload)),
+  };
+
+  const invokeCommand = new InvokeCommand(options);
+  const goInvoke = await go(() => awsLambdaClient.send(invokeCommand));
+  if (!goInvoke.success) {
+    logger.debug(`Lambda invocation or execution failed. ${goInvoke.error}}`);
+    throw goInvoke.error;
+  }
+  const invokeOutput = goInvoke.data;
+
+  if (invokeOutput.FunctionError) {
+    logger.debug(`Lambda invocation or execution failed. Response: ${invokeOutput.Payload}`);
+    throw invokeOutput.Payload;
+  }
+
+  if (!invokeOutput.Payload) {
+    const error = new Error('Missing payload in lambda response');
+    logger.debug(`Lambda invocation or execution failed. ${error}`);
+    throw error;
+  }
+
+  const payloadString = new TextDecoder('utf-8').decode(invokeOutput.Payload);
```

Comment:
I noticed that you do not pass 'utf-8' during encode, but you pass it during encode... It feels wierd. Why not pass it to encode as well?

Reply:
Because `TextEncoder` encodes **only** to UTF-8 (https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder). There is no way to set a different encoding. But `TextDecoder` can decode from multiple encodings (https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder).